File: server/src/Models/DMToolkitModels/Instances/SubclassInstance.cs
using DMToolkit.API.Models.DMToolkitModels.Definitions;

namespace DMToolkit.API.Models.DMToolkitModels.Instances;

public class SubclassInstance : IDefinitionInstance<SubclassDefinition>
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public ICollection<FeatInstance> FeatInstances { get; set; } = new List<FeatInstance>();
    public string DefinitionId { get; set; } = string.Empty;
    public SubclassDefinition? Definition { get; set; } = null;
    public string CharacterClassInstanceId { get; set; } = string.Empty;
    public CharacterClassInstance? CharacterClassInstance { get; set; } = null;
}

File: web-client/src/lib/types/dm-tool-types/instances/subclassInstance.ts
import type { SubclassDefinition } from "../definitions/subclassDefinition";
import type { FeatInstance, FeatInstanceDto } from "./featInstance";

export interface SubclassInstance {
    id: string;
    featInstances: FeatInstance[];
    definition: SubclassDefinition;
}

export interface SubclassInstanceDto {
    id: string;
    featInstances: FeatInstanceDto[];
    definitionId: string;
}